anxcloud/data_source_vlan.go

```diff
@@ -0,0 +1,111 @@
+package anxcloud
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"go.anx.io/go-anxcloud/pkg/vlan"
+)
+
+func dataSourceVLAN() *schema.Resource {
+	return &schema.Resource{
+		Description: "Provides details about an Anexia Cloud VLAN. This data source is useful if you want to use a non-terraform managed VLAN.",
+		ReadContext: dataSourceVLANRead,
+
+		// TODO: can I reuse the existing schema?
```

Comment:
we do have some helper functions to override field behaviors (see https://github.com/anexia-it/terraform-provider-anxcloud/blob/main/anxcloud/utils.go#L18)
but if the schemas are "too different", I tend to define them separately